python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -504,3 +510,39 @@ def get_asset_check_executions(
         but do not have a target materialization yet (since we don't set the target until the check is executed).
         """
         raise NotImplementedError()
+
+    @abstractmethod
+    def get_materialization_records(
+        self,
+        records_filter: Optional[Union[AssetKey, AssetRecordsFilter]] = None,
+        limit: Optional[int] = None,
+        ascending: bool = False,
+    ) -> Sequence[EventLogRecord]:
```

Comment:
how do users get valid cursors to pass in if we only return a sequence of events? Especially when dealing with `RunShardedEventsCursor`

Reply:
the `EventLogRecord` has the storage id on them.

Reply:
Oh.  I guess it could return a sequence of records and an explicit cursor?